web_src/js/components/ScopedAccessTokenSelector.vue

```diff
@@ -0,0 +1,102 @@
+<template>
+  <div class="scoped-access-token-category">
+    <div class="field gt-pl-2">
+      <div class="ui checkbox">
+        <input ref="category" v-model="categorySelected" class="enable-system" type="checkbox" name="scope" :value="'delete:' + category" @change="onCategoryInput">
+        <label>{{ category }}</label>
+      </div>
+    </div>
+    <div class="field gt-pl-4">
+      <div class="field">
+        <div class="ui checkbox">
+          <input ref="read" v-model="readSelected" :disabled="categorySelected || writeSelected" class="enable-system" type="checkbox" name="scope" :value="'read:' + category" @change="onIndividualInput">
+          <label>read:{{ category }}</label>
+        </div>
+      </div>
+      <div class="field">
+        <div class="ui checkbox">
+          <input ref="write" v-model="writeSelected" :disabled="categorySelected" class="enable-system" type="checkbox" name="scope" :value="'write:' + category" @change="onIndividualInput">
+          <label>write:{{ category }}</label>
+        </div>
+      </div>
+      <div class="field">
+        <div class="ui checkbox">
+          <input ref="delete" v-model="deleteSelected" :disabled="categorySelected" class="enable-system" type="checkbox" name="scope" :value="'delete:' + category" @change="onIndividualInput">
+          <label>delete:{{ category }}</label>
+        </div>
```

Comment:
`ui checkbox` only works for Fomantic UI checkbox in templates.

For Vue, it doesn't work well. I think it could be simple `<label><input></label>` here

Reply:
It looks like we'll lose some of the styling that way: 
<img width="332" alt="Screen Shot 2023-05-17 at 10 09 24 AM" src="https://github.com/go-gitea/gitea/assets/23248839/3f1744a8-ee8b-46cd-b4c9-44ab1e2cb1d2">

Reply:
No worry, there are already some PRs doing so:

* #23596

It won't be a problem.

Reply:
That PR was only for the issue list, not globally which is still TODO.

Reply:
It really needs some framework level support now.

Otherwise, `ui checkbox` in Vue would make the label part can't toggle the checkbox, which is very annoying for users.

Reply:
> Is nesting checkbox inside label not an option?

IMO "nesting checkbox inside label" is the proper way.

> How about a for attribute?

1. It would conflict a lot
2. It just introduces unnecessary IDs and makes code hard to read or maintain.

Reply:
In history , there were commits like this:

https://github.com/go-gitea/gitea/compare/583d5738b53673743672d8b8860f73d1ba9f50fb..2f2450f316f13ee6085be9da1cb2a16d28e4e5ad

(this commit revers unnecessary changes in #22593 

Manually adding ID/FOR attributes, I have strong objection for doing so because it's unmaintainable.

So a new PR replaces it: https://github.com/go-gitea/gitea/pull/22599

Reply:
Yes, nesting is better if possible. If not, we could autogenerate via something similar to react [`useId`](https://react.dev/reference/react/useId).